f8a_worker/workers/recommender.py

```diff
@@ -607,6 +608,39 @@ def _get_stack_values(self, similar_stacks_list):
             similarity_list.append(s_stack)
         return similarity_list
 
+def apply_license_filter(epv_list):
+    dependencies = []
+    licenses = []
+    license_score_list = []
+    for epv in epv_list:
+        name = epv.get('pkg', {}).get('name', [''])[0]
+        version = epv.get('ver', {}).get('version', [''])[0]
+        licenses = epv.get('ver', {}).get('version', {}).get('licenses', [])
+
+        license_scoring_input = {
+            'package': name,
+            'version': version,
+            'licenses': licenses
+        }
+        license_score_list.append(license_scoring_input)
+
+    # Call License Scoring to Get Stack License
+    license_analysis, dependencies = perform_license_analysis(license_score_list, dependencies)
+    conflicting_companions = []
+    if license_analysis['status'] == 'StackLicenseConflict':
+        conflict_packages = license_analysis.get('conflict_packages', {})
+        for conflict in conflict_packages:
+            for pkg in conflict.keys():
+                conflicting_companions.append(pkg)
+    conflicting_companions = list(set(conflicting_companions))
```

Comment:
why do you have to convert this to `list`?

Reply:
I guess it's to remove duplicates from list

```
In [1]: l=[1,2,1,4,3]

In [2]: list(set(l))
Out[2]: [1, 2, 3, 4]
```

Reply:
Even though you are right if you check my previous comment  `conflicting_companions` is defined as a `list`. Which is completely unnecessary, since it can be a `set` at first place so we don't have to use this conversion.

Reply:
For JSON conversion purposes, I have kept data in list and uses set() for deduplication.